examples/interrupts/device.rs

```diff
@@ -0,0 +1,52 @@
+use std::sync::{Arc, Mutex};
```

Comment:
Do you mind dropping this commit from the initial PR? It is not required, and it might be better to first focus on the implementation, and then add the examples only once the interfaces are stabilized.

Reply:
I actually found the example to be useful for understanding the whole picture.

Reply:
Yes, I agree they're useful for understanding how the abstractions work, but I'd like to avoid getting too many comments on a single PR. The examples can be linked in the PR (as the usage in Firecracker is), and it would also go in line with the [recommendation of keeping the PRs under 250 lines](https://hugooodias.medium.com/the-anatomy-of-a-perfect-pull-request-567382bb6067#:~:text=Recap-,Pull%20request%20size,of%20250%20lines%20of%20change.)). We can also just ignore them in the first iterations of the PR, and just review them after the interfaces are stabilized.

Reply:
I see what you mean. I think we can ignore the example in the first iterations to focus on the interfaces, and address them later on. @alsrdn would that be ok with you?

Reply:
yup, no problem.

Reply:
Removed the example from this PR. Will add it later.